environments/gcp/terraform.tf

```diff
@@ -0,0 +1,37 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 4.41.0"
+    }
+
+     kubectl = {
+      source  = "alekc/kubectl"
+      version = ">= 2.0.2"
+    }
+
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = ">= 4.41.0"
+    }
+    helm = {
+      source = "hashicorp/helm"
+      version = ">=2.11.0"
+  }
+
+  kubernetes = {
+      source = "hashicorp/kubernetes"
+      version = ">=2.23.0"
+    }
```

Comment:
Is this still used? Maybe I'm overlooking the respective resource.

Personal opinion:
I prefer this provider over the `kubectl` provider and use [`k8s2tf`](https://github.com/sl1pm4t/k2tf) to convert raw manifests to resources for the kubernetes provider.
Only reason for me to use the `kubectl` provider would be to avoid the  [chicken egg problem](https://www.reddit.com/r/Terraform/comments/uwn8m9/do_you_use_the_kubernetes_terraform_provider/) terraform unfortunatly suffers from when either the cluster doesn't exist yet or doesn't have certain CRDs installed at `terraform plan` time.
Does the chicken and egg problem occur here when trying to create the cluster, install cert manager, and create the `Issuer` resource at the same time?

Reply:
I will create a new Issue to track this

Reply:
>Does the chicken and egg problem occur here
For the most part, yes.

`cert-manager` itself requires several CRDs available before installation time, at least, for production environments. Trying to enable a `depends_on` with the GKE cluster while using `kubernetes_manifest` leads to a situation where the kubeconfig needs to be populated from an existing GKE cluster. This, in turn, can lead to inconvenient hacks like having two planning phases (one to build the cluster and another to apply the resources to it). 

In summary, that provider is effectively not being used and will be removed.

Reply:
It turns out it's been used to create a ns and a secret. A further step would be just submitting a single manifest to do both and getting rid of this provider. Leaving it for a further release (minor effort)